card/mod/01_core/set/all/name.rb

```diff
@@ -31,10 +31,14 @@ def name= newname
 
   newkey = cardname.key
   if key != newkey
+    was_in_cache = Card.cache.soft.delete key
     self.key = newkey
+    # keep the soft cache up-to-date
+    Card.write_to_soft_cache self if was_in_cache
```

Comment:
good point (developed right in the battle against hordes of failing wikirate tests - no time for pretty solutions ;)